src/libfmt_macros/lib.rs

```diff
@@ -259,9 +262,41 @@ impl<'a> Parser<'a> {
     /// Parses an Argument structure, or what's contained within braces inside
     /// the format string
     fn argument(&mut self) -> Argument<'a> {
+        let mut pos = self.position();
+        let mut format = self.format();
+
+        // Resolve CountIsNextParam's into absolute references.
+        // Current argument's position must be known so this is done after
+        // format parsing.
+        // Curiously, currently {:.*} for named arguments is implemented,
+        // and it consumes a positional arg slot just like a positional {:.*}
+        // does. The current behavior is reproduced to prevent any
+        // incompatibilities.
+        match format.precision {
+            CountIsNextParam => {
+                // eat the current implicit arg
+                let i = self.curarg;
+                self.curarg += 1;
+                format.precision = CountIsParam(i);
+            }
+            _ => {}
+        }
```

Comment:
I wonder if it might be best to just remove the `CountIsNextParam` and `ArgumentNext` enum variants? That way this could be handled directly in the `position` and `format` methods